### ISSUE TITLE
Update<T,K>

### ISSUE THREAD

**Issue by NeilMenne** [elastic/elasticsearch-net#732]
So I'm trying to pass a partial update this way (http://stackoverflow.com/a/23796603) in a test as well as in the production code that utilizes Nest. When I do this, the test never completes execution. It hangs indefinitely in the `Client.Update<Load, object>` function.

The reason I began investigating this approach was I was trying to use a new instance of `Load` to do the partial update and information in the `Total` field was lost.

My fix in the meantime is to use nullable fields for those that get overwritten i.e. `int?` instead of `int` and `double?` instead of `double`.

A simplified version of my class:

```
[ElasticType(Name = "load",
    IdProperty = "Id")]
public class Load
{
    public int Id { get; set; }

    public int CustomerId { get; set; }

    public string CustomerName { get; set; }

    public double Total { get; set; }
}
```

**tl;dr: Unexpected behavior for `Update<T,K>` when `Document` is given a dynamic object**
